store/postgres/migrations/2021-05-20-130220_graph_node_version/up.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE IF NOT EXISTS subgraphs.graph_node_versions (
+    id serial PRIMARY KEY,
+    git_commit_hash text NOT NULL,
+    git_repository_dirty boolean NOT NULL,
+    crate_version text NOT NULL,
+    major integer NOT NULL,
+    minor integer NOT NULL,
+    patch integer NOT NULL,
+    pre_release text NOT NULL,
+    rustc_version text NOT NULL,
+    rustc_host text NOT NULL,
+    rustc_channel text NOT NULL,
+    CONSTRAINT unique_graph_node_versions UNIQUE (git_commit_hash, git_repository_dirty, crate_version, major, minor, patch, pre_release, rustc_version, rustc_host, rustc_channel)
```

Comment:
I'm a little bit concerned about limiting the constraint to only `(major, minor patch)` columns, as this will make it possible for us to store states different from reality. I assumed that every column on this table is meaningful as the version number trio, so any column not uniquely indexed could potentially prevent the desired insert.

For instance, if we already have a record for graph node version `0.22.0` for `git_commit_hash = A,` then it would be impossible to insert another record for the same version but having `git_commit_hash = B`, or any different field that is not listed within the `unique` constraint.

Reply:
That seems to be an important data modeling question: I assumed that major/minor/patch uniquely identifies all the other data. For release builds that should be true. `crate_version` as I understand it is just major/minor/patch in a different format. I think having the same commit hash for different versions is not something to worry about, I don't think we would ever do that.

The one thing where this doesn't fit well is during development since you could have the same version with different builds. But maybe for those, it's fine to just ignore any duplication.